README
NAME
    Dist::Zilla::Plugin::ReportPhase - Report whats going on.

VERSION
    version 0.02

SYNOPSIS
    In your dist.ini:

      [ReportPhase / Phase_Begins]

      ...

      [ReportPhase / Phase_Ends]

DESCRIPTION
    This plugin was written to give the author some idea of the order that
    various roles are invoked under different conditions. So, ReportPhase
    implements every major role, and reports when it is being invoked. For
    best results it should be used as both the very first, and the very last
    plugin listed in dist.ini, as it will then report entering and exiting
    every phase.

    Other than this phase reporting, this plugin has no use.

ATTRIBUTES
    NONE

AUTHOR
    Stirling Westrup <swestrup@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2010 by Stirling Westrup.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



CHANGES
Revision History for Dist-Zilla-Plugin-ReportPhase
==================================================

v0.02 2013-02-12
----------------

* Date: 2013-02-12 18:02:12 -0500
  Commit: Stirling Westrup <stirling@userful.ca>
  Hash: c7b28137c57de995885f6666f5607ab1537fbf92

    Fix dist.ini so that it sets the minimum perl needed correctly


v0.01 2013-02-12
----------------

* Date: 2013-02-12 17:21:51 -0500
  Commit: Stirling Westrup <stirling@userful.ca>
  Hash: 58ce0d34ffce57c739d7101d941bf96bcdab10c1

    Update ReportPhase to demand perl 5.006 minimum.

* Date: 2013-02-12 17:18:28 -0500
  Commit: Stirling Westrup <stirling@userful.ca>
  Hash: d5553d1c43c9bd4c4b7af06afff458fa94303289

    Updated dist.ini to remove critic tests we can't fix

* Date: 2013-02-12 17:04:56 -0500
  Commit: Stirling Westrup <stirling@userful.ca>
  Hash: 598f796f88ca648a51fe65f2339136617c3e2395

    Updated dist.ini to replace deprecated plugins

* Date: 2013-02-12 16:54:08 -0500
  Commit: Stirling Westrup <stirling@userful.ca>
  Hash: 059a0ea0681d4eb911ff9f85ac534afcacce2265

    Fixed logging issue


===================================
End of changes in the last 365 days
===================================


ReportPhase.pm
package Dist::Zilla::Plugin::ReportPhase;
# ABSTRACT: Log every role use in every phase executed.
BEGIN
  {
    $Dist::Zilla::Plugin::ReportPhase::VERSION
      = substr '$$Version: 0.02 $$', 11, -3;
  }

use 5.006;
use Moose;
use Moose::Autobox;
use Data::Dumper;

with
  ( 'Dist::Zilla::Role::AfterBuild'
  , 'Dist::Zilla::Role::AfterMint'
  , 'Dist::Zilla::Role::AfterRelease'
  , 'Dist::Zilla::Role::BeforeArchive'
  , 'Dist::Zilla::Role::BeforeBuild'
  , 'Dist::Zilla::Role::BeforeMint'
  , 'Dist::Zilla::Role::BeforeRelease'
  , 'Dist::Zilla::Role::BuildRunner'
  , 'Dist::Zilla::Role::ConfigDumper'
  , 'Dist::Zilla::Role::FileFinder'
  , 'Dist::Zilla::Role::FileGatherer'
  , 'Dist::Zilla::Role::FileMunger'
  , 'Dist::Zilla::Role::FilePruner'
  , 'Dist::Zilla::Role::InstallTool'
  , 'Dist::Zilla::Role::MetaProvider'
  , 'Dist::Zilla::Role::MintingProfile'
  , 'Dist::Zilla::Role::ModuleMaker'
  , 'Dist::Zilla::Role::PrereqSource'
  , 'Dist::Zilla::Role::Releaser'
  , 'Dist::Zilla::Role::ShareDir'
  , 'Dist::Zilla::Role::TestRunner'
  , 'Dist::Zilla::Role::VersionProvider'
  );


my $cmd = ref $App::Cmd::active_cmd;

sub _report{ $_[0]->logger->log("########## $_[1] ##########"); }

sub after_build      { $_[0]->_report("After Build");       }
sub after_mint       { $_[0]->_report("After Mint");	    }
sub after_release    { $_[0]->_report("After Release");     }
sub before_archive   { $_[0]->_report("Before Archive");    }
sub before_build     { $_[0]->_report("Before Build");      }
sub before_mint      { $_[0]->_report("Before Mint");       }
sub before_release   { $_[0]->_report("Before Release");    }
sub build	     { $_[0]->_report("Build");		    }
sub bundle_config    { $_[0]->_report("Bundle Config");	    }
sub dump_config      { $_[0]->_report("Dump Config");       }
sub find_files       { $_[0]->_report("Find Files");	    }
sub gather_files     { $_[0]->_report("Gather Files");      }
sub make_module      { $_[0]->_report("Module Maker");	    }
sub metadata	     { $_[0]->_report("Metadata");	    }
sub munge_files      { $_[0]->_report("Munge Files");       }
sub munge_file       { $_[0]->_report("Munge File: @{[$_[1]->name]}"); }
sub profile_dir      { $_[0]->_report("Profile Dir");	    }
sub provide_version  { $_[0]->_report("Provide Version");   }
sub prune_files      { $_[0]->_report("Prune Files");       }
sub register_prereqs { $_[0]->_report("Bundle Config");	    }
sub release	     { $_[0]->_report("Release");	    }
sub setup_installer  { $_[0]->_report("Setup Installer");   }
sub share_dir_map    { $_[0]->_report("Share Dir");	    }
sub test	     { $_[0]->_report("Test Runner");       }
sub configure	     { $_[0]->_report("Configure");	    }
sub dir		     { $_[0]->_report("Dir");		    }

__PACKAGE__->meta->make_immutable;
no Moose;
1;

__END__
=pod

=head1 NAME

Dist::Zilla::Plugin::ReportPhase - Report whats going on.

=head1 VERSION

version 0.02

=head1 SYNOPSIS

In your F<dist.ini>:

  [ReportPhase / Phase_Begins]

  ...

  [ReportPhase / Phase_Ends]

=head1 DESCRIPTION

This plugin was written to give the author some idea of the order that
various roles are invoked under different conditions. So,
B<ReportPhase> implements every major role, and reports when it is
being invoked. For best results it should be used as both the very
first, and the very last plugin listed in dist.ini, as it will then
report entering and exiting every phase.

Other than this phase reporting, this plugin has no use.

=head1 ATTRIBUTES

NONE

=head1 AUTHOR

Stirling Westrup <swestrup@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2010 by Stirling Westrup.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Log every role use in every phase executed.",
  "AUTHOR" => "Stirling Westrup <swestrup\@gmail.com>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-ReportPhase",
  "EXE_FILES" => [],
  "LICENSE" => "gpl",
  "NAME" => "Dist::Zilla::Plugin::ReportPhase",
  "PREREQ_PM" => {},
  "VERSION" => "0.02",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



